merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresResultsCellRepository.java

```diff
@@ -70,6 +70,32 @@ public PostgresResultsCellRepository(final DataSource dataSource) {
     }
   }
 
+  @Override
+  public Optional<ResultsProtocol.OwnerRole> claim(final PlanId planId, final Long datasetId) {
+    try (final var connection = this.dataSource.getConnection()) {
+      claimSimulationDataset(connection, datasetId);
+      System.out.printf("Claimed simulation with datatset id %s%n", datasetId);
+
+      final var planStart = getPlan(connection, planId).startTime();
+
+      final var simulation$ = getSimulation(connection, planId);
+      if (simulation$.isEmpty()) {
+        return Optional.empty();
+      }
+      final var simulation = simulation$.get();
+
+      return Optional.of(new PostgresResultsCell(
+          this.dataSource,
+          simulation,
+          datasetId,
+          planStart));
+    } catch (final NoSuchPlanException ex) {
+      throw new Error(String.format("Cannot process simulation for nonexistent plan %s.%n", planId), ex);
+    } catch(final SQLException | DatabaseException ex) {
+      return Optional.empty();
```

Comment:
Given the discussion with @kroffo and @mattdailis I've adjusted both the `ClaimSimulationAction` and the `PostgresResultsCellrepository` which uses it. The action now only returns an SQLException when an unexpected number of result rows are returned by the db update. An `UnclaimableSimulationException` has been added, although no action is taken because of it so I'm unsure if this is the best remedy here. Still pondering...

Reply:
I think this solution adequately addresses @kroffo 's concerns about using the `SQLException` to mean two different things :+1:

I would maybe request a javadoc comment on `claimSimulationDataset` and/or `ClaimSimulationAction.apply` explaining what the `UnclaimableSimulationException` means.

Reply:
Some thoughts that went through my head were that if this method were returning some data that the caller needed, we could return an Optional and feel reasonably certain that the caller will use that information.

In this case, the whole point of `claim` is its side-effect - the caller just needs to check whether it succeeded or not. Declaring the checked exception as part of the method signature forces the caller to handle the failure case.

We could potentially contort the data structures to make `claim` return some data that the caller needed (e.g. caller only has `datasetId`, in order to get the rest of the information they call `claim`. Thus if `claim` returns an `Optional.empty`, the caller cannot erroneously proceed and try to run the simulation), but I'm not sure I like that either.

If our external advisory board of @Twisol or @kroffo (or anyone else who has opinions on when to use exceptions vs when to return a success/failure value) wants to weigh in, feel free :+1:

Reply:
Good idea to add some Javadocs. I did so.

Reply:
I haven't looked at the code (GitHub is being very unhelpful; it says "these commits no longer exist"), but not being able to claim a simulation just means the worker should go back to the top and wait for another simulation. It probably couldn't claim because someone already claimed it.

Based on what Matt described just above, I would argue for an `Optional<>` based approach, where an object permitting write access to the simulation is provided only if the claim was successful.

Reply:
At the ResultsCellRepository interface level, that's exactly how it works (I think): 

https://github.com/NASA-AMMOS/aerie/blob/11751aaa624f97fa1fbab5f56a91ee694f12d0b8/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/ResultsCellRepository.java#L11

The implementation of the `PostgresResultCellRepository#claim` converts an `UnclaimableSimulationException` into an Optional:

https://github.com/NASA-AMMOS/aerie/blob/11751aaa624f97fa1fbab5f56a91ee694f12d0b8/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresResultsCellRepository.java#L90-L115

I think I was musing about the `ClaimSimulationAction`, which is a `void` method that throws an `UnclaimableSimulationException`:

https://github.com/NASA-AMMOS/aerie/blob/11751aaa624f97fa1fbab5f56a91ee694f12d0b8/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/ClaimSimulationAction.java#L24

I guess I'm unsure whether it's appropriate for the `UnclaimableSimulationException` to exist. My current stance is that since its local to the repository, it's fine to throw an exception - but I don't feel like I have a good sense of when it makes sense to use an exception, and when a "failure" should be part of the return type.